mars/services/scheduling/supervisor/autoscale.py

```diff
@@ -0,0 +1,269 @@
+            await self._cluster_api.set_node_status(
+                node=address, role=NodeRole.WORKER, status=NodeStatus.STOPPING)
+        # Ensure global_slot_manager get latest bands timely, so that we can invoke `is_band_idle`
+        # to ensure there won't be new tasks scheduled to the stopping worker.
+        await self.global_slot_ref.refresh_bands()
+        excluded_bands = set(b for bands in workers_bands.values() for b in bands)
+
+        async def release_worker(address):
+            logger.info("Start to release worker %s.", address)
+            worker_bands = workers_bands[address]
+            for band in worker_bands:
+                while not await self.global_slot_ref.is_band_idle(band):
+                    await asyncio.sleep(0.1)
+            await self._migrate_data_of_bands(worker_bands, excluded_bands)
+            await self._cluster_api.release_worker(address)
+            self._dynamic_workers.remove(address)
+            logger.info("Released worker %s.", address)
+
+        await asyncio.gather(*[release_worker(address) for address in addresses])
+
+    def get_dynamic_workers(self) -> Set[str]:
+        return self._dynamic_workers
+
+    def get_dynamic_worker_nums(self) -> int:
+        return len(self._dynamic_workers)
+
+    async def get_worker_bands(self, worker_address) -> List[BandType]:
+        node_info = (await self._cluster_api.get_nodes_info(
+            [worker_address], resource=True, exclude_statuses=set()))[worker_address]
+        return [(worker_address, resource_type) for resource_type in node_info['resource'].keys()]
+
+    async def _migrate_data_of_bands(self, bands: List[BandType], excluded_bands: Set[BandType]):
+        """Move data from `bands` to other available bands"""
+        session_ids = list(self.queueing_refs.keys())
+        for session_id in session_ids:
+            from mars.services.meta import MetaAPI
+            meta_api = await MetaAPI.create(session_id, self.address)
+            for src_band in bands:
+                band_data_keys = await meta_api.get_band_chunks(src_band)
+                for data_key in band_data_keys:
+                    dest_band = await self._select_target_band(src_band, data_key, excluded_bands)
+                    # For ray backend, there will only be meta update rather than data transfer
+                    try:
+                        await (await self._get_storage_api(session_id, dest_band[0])).fetch(
+                            data_key, band_name=src_band[1], remote_address=src_band[0])
+                    except Exception:
+                        logger.info('chunk meta %s', await meta_api.get_chunk_meta(data_key))
+                        raise
+                    await (await self._get_storage_api(session_id, src_band[0])).delete(data_key)
+                    chunk_bands = (await meta_api.get_chunk_meta(data_key, fields=['bands'])).get('bands')
+                    chunk_bands.remove(src_band)
```

Comment:
Utilize `remove_chunk_band` (or a straightforward `remove_band_for_all_data`) to reduce number of calls.